flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumerator.java

```diff
@@ -407,7 +407,12 @@ public PartitionOffsetsRetrieverImpl(
                         .thenApply(
                                 result -> {
                                     Map<TopicPartition, Long> offsets = new HashMap<>();
-                                    result.forEach((tp, oam) -> offsets.put(tp, oam.offset()));
+                                    result.forEach(
```

Comment:
Yes, this test is change is verified by `OffsetsInitializerTest#testSpecificOffsetsInitializer`, which is updated by this PR.

The test would call `committedOffsets(partitions)` where `partitions` include a partition that does not have committed offsets.

I verified that the test `testSpecificOffsetsInitializer` would fail if we don't update the `committedOffsets(...)` method here.